app/logic/questionnaire_translator.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class QuestionnaireTranslator
+  class << self
+    def translate_content(content, locale)
+      @locale = locale
+      translate_content_aux(content)
+    end
+
+    private
+
+    def translate_content_aux(content)
+      return content if content.blank?
+
+      if content.is_a?(Hash)
+        # replace with translation
+        return translate_content_aux(content[@locale]) if content.key?(@locale)
+
+        return translate_content_aux(content[@locale.to_sym]) if content.key?(@locale.to_sym)
+
+        content.keys.each do |key|
+          content[key] = translate_content_aux(content[key])
+        end
+        return content
+      end
+      if content.is_a?(Array)
+        content.each_with_index do |value, idx|
```

Comment:
can't you do a `return content.map{|x| translate_content_aux(x)}` here? It might not work well with the recursion, not sure.

Reply:
Yes, I think that's equivalent.

Reply:
I liked explicitly assigning things to content, but since we're returning that value as well, we don't need to modify it in place.

Reply:
Fixed.